VP-fitness-frontend/app/src/main/java/com/example/neil/vp_fitness_frontend/OnboardingFragmentActivity.java

```diff
@@ -0,0 +1,230 @@
+            public void onPageSelected(int position) {
+            }
+
+            @Override
+            public void onPageScrollStateChanged(int state) {
+
+            }
+        });
+    }
+
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_onboarding, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+        pageId = id;
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    /**
+     * A placeholder fragment containing a simple view.
+     */
+    public static class PlaceholderFragment extends Fragment {
+        /**
+         * The fragment argument representing the section number for this
+         * fragment.
+         */
+        private static final String ARG_SECTION_NUMBER = "section_number";
+
+        public PlaceholderFragment() {
+        }
+
+        /**
+         * Returns a new instance of this fragment for the given section
+         * number.
+         */
+        public static PlaceholderFragment newInstance(int sectionNumber) {
+            PlaceholderFragment fragment = new PlaceholderFragment();
+            Bundle args = new Bundle();
+            args.putInt(ARG_SECTION_NUMBER, sectionNumber);
+            fragment.setArguments(args);
+            return fragment;
+        }
+
+        @Override
+        public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                                 Bundle savedInstanceState) {
+            View rootView = inflater.inflate(R.layout.tab1, container, false);
+            return rootView;
+        }
+    }
+
+    /**
+     * Method to call when finished with the onboarding process.
+     */
+    public void finish(View view) {
+
+        OnboardingFragment tab = (OnboardingFragment) fragments.get("TAB3");
+        Log.e("RESULTS", tab.getData().toString());
+
+    }
+
+    /**
+     * A {@link FragmentPagerAdapter} that returns a fragment corresponding to
+     * one of the sections/tabs/pages.
+     */
+    public class SectionsPagerAdapter extends FragmentPagerAdapter {
+
+        public SectionsPagerAdapter(FragmentManager fm) {
+            super(fm);
+        }
+
+
+        @Override
+        public Fragment getItem(int position) { // getItem is called to instantiate the fragment for the given page.
+
+            // Switch statement for tabs in onboarding fragment
+            switch (position) {
+                case 0:
+                    Tab1 tab1 = new Tab1();
```

Comment:
Right now these are all named Tab1, Tab2, etc... These should instead be renamed to their function, i.e. `OnboardAccountInfoTab`, `OnboardIntroTab`, `OnboardEquipmentTab`. Seems tedious but will be better for debugging and moving things around if needed.